### ISSUE TITLE
The Antd Select scrolls a page not a drop down

### ISSUE THREAD

**Issue by EgorPashko** [ant-design/ant-design#37674]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/rough-firefly-5l1zol?file=/package.json:230-236)

### Steps to reproduce

1) Open code sandbox on mobile (I use safari)
2) Open drop down by clicking on the select box
3) Try scroll inside a dropdown 

### What is expected?

When I scroll inside drop down, it scrolls inside drop down

### What is actually happening?

When I scroll inside drop down, sometimes it scrolls the page

| Environment | Info |
| --- | --- |
| antd | 4.23.2 |
| React | 18.0.0 |
| System | Mobile |
| Browser | Safari |

---
When trying to scroll inside the dropdown menu ONLY on mobile sometimes it scrolls the page.
I use react, but not sure that is related to any FE libraries/frameworks IOS Safari,
Body and html do not have any styles

https://www.youtube.com/shorts/NXs3ou8twHQ

**Comment by zombieJ** [comment]
Could you help to record the touch point in video also? Seems it work at the begin but failed after a while.

**Comment by EgorPashko** [comment]
@zombieJ sure
Created a new video. https://www.youtube.com/shorts/it31nipjy-s

**Comment by jeff-jacobsen** [comment]
This is still happening, would love it to be fixed

**Comment by WiktorG** [comment]
I'm having the same issue.

**Comment by sc4lr** [comment]
+1
Would love this to be fixed

**Comment by KuznetsovCreator** [comment]
+3
looking forward for this bug to be eliminated @zombieJ

**Comment by joelalejandro** [comment]
Antd uses `react-component/select` - https://github.com/react-component/select/

Just found this issue, and apparently setting `virtual={false}` fixes the problem, although that means you cannot use virtual list rendering.

**Comment by KuznetsovCreator** [comment]
> Antd uses `react-component/select` - https://github.com/react-component/select/
> 
> Just found this issue, and apparently setting `virtual={false}` fixes the problem, although that means you cannot use virtual list rendering.

That's right, and it's a problem for dropdown lists with a lot of items. For example, for a list of 600+ cities.

**Comment by onlyMe7023** [comment]
@joelalejandro thanks man it's working 😀😀

**Comment by gherciu** [comment]
Any news on this I'm facing same issue even in last version
@zombieJ

**Comment by gherciu** [comment]
even with virtual={false} the scroll is kinda buggy, it randomly stops, at some item
Why not use native scroll if virtual is turned off?

**Comment by OrenMoveo** [comment]
> even with virtual={false} the scroll is kinda buggy, it randomly stops, at some item
> Why not use native scroll if virtual is turned off?

Facing the exact same issue, any fixes?

**Comment by nadavhalfon** [comment]
Same problem, feels very buggy on mobile

**Comment by nadavhalfon** [comment]
I have a bit of a hacky solution but it works.
This is the problem and I'll explain

<img width="606" alt="Image" src="https://github.com/user-attachments/assets/361de616-1f9a-45e6-abe8-0522adede5e1" />

At first I thought the highlighted div is causing the problem - it's parent 'rc-virtual-list-holder' is the one that has the max height and overflow style, so instead of scrolling the items in the list below if you track the div movement you'll se we kind of scroll a large big div up and down. but when trying by editing the dom manually I couldn't find the exact problem but this one works:

```
<Select 
   virtual={false}
   {...otherPropsAsYouNeed}
   options={someArray}
   dropdownRender={(menu) => (
    <div className="h-[250px] overflow-auto">
      {customerSearchValue && (
        <>
          {customComponentOfMine}
          <Divider className="my-1" />
        </>
      )}
      {menu}
    </div>
  )}
/>
```
And you must also add this CSS (you can use styled-components if you want):
```
.rc-virtual-list-holder {
  max-height: unset !important;
  overflow-y: unset !important;
}
```

Now iOS scrolling is awesome :)
